File: SHSegmentedControlTableView/Demo/ViewController.m
//
//  ViewController.m
//  SHSegmentedControlTableView
//
//  Created by angle on 2017/10/10.
//  Copyright © 2017年 angle. All rights reserved.
//

#import "ViewController.h"


@interface ViewController ()<UITableViewDataSource,UITableViewDelegate>

@property (nonatomic, strong) UITableView *tableView;

@property (nonatomic, strong) NSArray *dataArray;

@property (nonatomic, strong) NSArray *controllerArray;

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    
    self.navigationItem.title = @"样式列表";
    
    self.dataArray = @[@"默认样式",@"涌入放大",@"右上角小标题",@"涌入放大+右上角小标题",@"导航栏透明",@"无头部",@"无bar",@"CollectionView",@"tableView+collectionView",@"下拉刷新全部",@"下拉刷新当前选中",@"独立上拉加载",@"头部固定-scrollView容器"];
    self.controllerArray = @[@"OneViewController",@"TwoViewController",@"ThreeViewController",@"FourViewController",@"FiveViewController",@"SixViewController",@"SevenViewController",@"EightViewController",@"NineViewController",@"TenViewController",@"TenViewController",@"ZeroViewController",@"ElevenViewController"];


    [self.view addSubview:self.tableView];
}
#pragma mark -
#pragma mark   ==============UITableViewDataSource==============
- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
    return 1;
}
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.dataArray.count;
}
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"cell" forIndexPath:indexPath];
    cell.textLabel.text = self.dataArray[indexPath.row];
    return cell;
}
#pragma mark -
#pragma mark   ==============UITableViewDelegate==============
- (CGFloat)tableView:(UITableView *)tableView heightForRowAtIndexPath:(NSIndexPath *)indexPath {
    return 45.f;
}
- (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath {
    [tableView deselectRowAtIndexPath:indexPath animated:YES];
    
    NSString *classStr = self.controllerArray[indexPath.row];
    UIViewController *controller = [[NSClassFromString(classStr) alloc] init];
    if ([classStr isEqualToString:@"TenViewController"] && indexPath.row == 9) {
        [controller setValue:@"1" forKey:@"all"];
    }
    if (controller) {
        controller.title = self.dataArray[indexPath.row];
        [self.navigationController pushViewController:controller animated:YES];
    }
}

#pragma mark -
#pragma mark   ==============UI-lazy==============
- (UITableView *)tableView {
    if (!_tableView) {
        _tableView = [[UITableView alloc] initWithFrame:self.view.bounds];
        _tableView.dataSource = self;
        _tableView.delegate = self;
        [_tableView setTableFooterView:[UIView new]];
        [_tableView registerClass:UITableViewCell.class forCellReuseIdentifier:@"cell"];
    }
    return _tableView;
}
#pragma mark -
#pragma mark   ==============memoryWarning==============
- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}


@end

File: Swift/SHSegmentedControlTableView/SHSegmentedControlTableView/ViewController.swift
//
//  ViewController.swift
//  SHSegmentedControlTableView
//
//  Created by angle on 2017/12/7.
//  Copyright © 2017年 angle. All rights reserved.
//

import UIKit


class ViewController: UIViewController, UITableViewDataSource, UITableViewDelegate {
    
    
    var dataArray: NSArray!
    var tableView: UITableView!
    var controllerArray: NSArray!

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        self.navigationItem.title = "样式列表"
        
        self.dataArray = ["默认样式", "涌入放大", "右上角小标题", "涌入放大+右上角小标题", "导航栏透明", "无头部", "无bar", "collectionView", "tableView+collectionView", "下拉刷新全部", "下拉刷新当前选中", "独立上拉加载", "头部固定-scrollView容器"]
        
        self.controllerArray = NSArray.init(objects: OneViewController(), TwoViewController(), ThreeViewController(), FourViewController(), FiveViewController(), SixViewController(), SevViewController(), EigViewController(), NineViewController(), TenViewController(),TenViewController(), ZeroViewController(), ElevViewController())
        self.tableView = self.getTableView()
        
        
        self.view.addSubview(self.tableView)
    }

    func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return self.dataArray.count
    }
    
    func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        let cell:UITableViewCell! = tableView.dequeueReusableCell(withIdentifier: "cell")
        cell.textLabel?.text = self.dataArray.object(at: indexPath.row) as? String
        return cell
    }
    
    func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
        tableView.deselectRow(at: indexPath, animated: true)
        let vc:SHBaseViewController = self.controllerArray.object(at: indexPath.row) as! SHBaseViewController
        if indexPath.row == 9 && vc.isKind(of: TenViewController.classForCoder()) {
            let tenVC:TenViewController = vc as! TenViewController
            tenVC.all = true
        }
        self.navigationController?.pushViewController(self.controllerArray.object(at: indexPath.row) as! UIViewController, animated: true)
    }
    
    func getTableView() -> UITableView {
        
        if self.tableView != nil {
            return self.tableView;
        }
        
        let tableView = UITableView.init(frame: self.view.bounds, style: UITableViewStyle.plain)
        tableView.dataSource = self
        tableView.delegate = self
        tableView.rowHeight = 45;
        tableView.register(UITableViewCell.classForCoder(), forCellReuseIdentifier: "cell")
        tableView.tableFooterView = UIView.init()
        return tableView
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

